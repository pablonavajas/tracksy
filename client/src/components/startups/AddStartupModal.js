import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { addStartup } from '../../actions/startupsActions';

import M from 'materialize-css/dist/js/materialize.min.js';

const AddStartupModal = ({ addStartup }) => {
  const [name, setName] = useState('');
  const [website, setWebsite] = useState('');
  const [ownership, setOwnership] = useState('');
  const [board, setBoard] = useState('');
  const [investment_1, setInvestment_1] = useState('');
  const [type_1, setType_1] = useState('');
  const [date_1, setDate_1] = useState('');
  const [investment_2, setInvestment_2] = useState('');
  const [type_2, setType_2] = useState('');
  const [date_2, setDate_2] = useState('');

  const onSubmit = () => {
    if (
      name === '' ||
      website === '' ||
      ownership === '' ||
      board === '' ||
      investment_1 === '' ||
      type_1 === '' ||
      date_1 === '' ||
      investment_2 === '' ||
      type_2 === '' ||
      date_2 === ''
    ) {
      M.toast({
        html: 'Please enter NA in the fields that are not applicable'
      });
    } else {
      const newStartup = {
        name,
        website,
        ownership,
        board,
        investment_1,
        investment_2,
        type_1,
        type_2,
        date_1,
        date_2
      };

      addStartup(newStartup);

      // Clear fields
      setName('');
      setWebsite('');
      setOwnership('');
      setBoard('');
      setInvestment_1('');
      setDate_1('');
      setType_1('');
      setInvestment_2('');
      setDate_2('');
      setType_2('');
    }
    //console.log('Submit button Clicked');
  };

  return (
    <div id="add-startup-modal" className="modal" style={modalStyle}>
      <div className="modal-content">
        <h4>Enter New Startup</h4>
        {/** Name */}
        <div className="row">
          <div className="input-field">
            <input
              id="name"
              type="text"
              name="name"
              value={name}
              onChange={e => setName(e.target.value)}
            />
            <label>Startup Name</label>
          </div>
        </div>

        {/** Website */}
        <div className="row">
          <div className="input-field">
            <input
              id="website"
              type="text"
              name="name"
              value={website}
              onChange={e => setWebsite(e.target.value)}
            />
            <label>Website</label>
          </div>
        </div>

        {/** Ownership */}
        <div className="row">
          <div className="input-field">
            <input
              id="ownership"
              name="ownership"
              type="text"
              value={ownership}
              onChange={e => setOwnership(e.target.value)}
            />
            <label>Ownership (%)</label>
          </div>
        </div>

        {/** Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              id="investment_1"
              name="investment_1"
              type="text"
              value={investment_1}
              onChange={e => setInvestment_1(e.target.value)}
            />
            <label>Investment 1</label>
          </div>
        </div>
        {/** Type of Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              id="type_1"
              name="type_1"
              type="text"
              value={type_1}
              onChange={e => setType_1(e.target.value)}
            />
            <label>Type of Investment 1</label>
          </div>
        </div>
        {/** Closing Date Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              id="date_1"
              name="date_1"
              value={date_1}
              type="date"
              onChange={e => setDate_1(e.target.value)}
            />
            <label>Close Date</label>
          </div>
        </div>

        {/** Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              id="investment_2"
              name="investment_2"
              type="text"
              value={investment_2}
              onChange={e => setInvestment_2(e.target.value)}
            />
            <label>Investment 2</label>
          </div>
        </div>

        {/** Type of Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              id="type_2"
              name="type_2"
              type="text"
              value={type_2}
              onChange={e => setType_2(e.target.value)}
            />
            <label>Type of Investment 2</label>
          </div>
        </div>

        {/** Closing Date Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              id="date_2"
              name="date_2"
              value={date_2}
              type="date"
              onChange={e => setDate_2(e.target.value)}
            />
            <label>Close</label>
          </div>
        </div>
      </div>
      <div className="row">
        <div className="input-field col s12">
          <select
            id="board"
            name="board"
            value={board}
            className="browser-default"
            onChange={e => setBoard(e.target.value)}
          >
            <option value="" disabled>
              Select Board
            </option>
            <option value="Member">Member</option>
            <option value=" - "> - </option>
            <option value="Observer"> Observer </option>
          </select>
        </div>
      </div>
      <div className="modal-footer">
        <a
          href="#!"
          onClick={onSubmit}
          className="modal-close waves-effect blue waves-light btn"
        >
          Submit
        </a>
      </div>
    </div>
  );
};

const modalStyle = {
  width: '60%',
  height: '75%'
};

AddStartupModal.propTypes = {
  addStartup: PropTypes.func.isRequired
};

export default connect(null, { addStartup })(AddStartupModal);
