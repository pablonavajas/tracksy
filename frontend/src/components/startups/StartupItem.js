import React from "react";
import PropTypes from "prop-types";
import CurrencyFormat from "react-currency-format";
var moment = require("moment");

import { connect } from "react-redux";
import { deleteStartup, setCurrent } from "../../actions/startupsActions";
import { ClickIcon } from "../smallComponents/displayComponents";
import { Chip } from "../smallComponents/displayComponents";

export const StartupItem = ({ startup, setCurrent, deleteStartup }) => {
  return (
    <tr>
      <th className="center">
        <a href={`https://${startup.website}`}>
          <img
            src={`http://logo.clearbit.com/${startup.website}`}
            alt=""
            width="100"
            height="100"
          />
        </a>
        <div>{startup.name}</div>
      </th>
      <td className="center">
        <Chip data={startup.ownership} />
      </td>
      <td className="center">
        <Chip data={startup.board} />
      </td>
      <td className="center">
        {startup.investments.map((investment) => (
          <div className="center" key={investment.id}>
            <Chip
              data={
                <CurrencyFormat
                  value={investment.value}
                  displayType={"text"}
                  thousandSeparator={true}
                  prefix={investment.currency}
                />
              }
            />
          </div>
        ))}
      </td>
      <td className="center">
        {startup.investments.map((investment) => (
          <div className="center" key={investment.id}>
            <Chip data={investment.investmentType} />
          </div>
        ))}
      </td>
      <td>
        {startup.investments.map((investment) => (
          <div className="center" key={investment.id}>
            <Chip data={moment(investment.date).format("DD/MM/YYYY")} />
          </div>
        ))}
      </td>
      <td>
        <ClickIcon
          href="#startup-overview"
          setFunction={setCurrent}
          val={startup}
          iconName="dehaze"
        />
      </td>
      <td>
        <ClickIcon
          href="#edit-startup-modal"
          setFunction={setCurrent}
          val={startup}
          iconName="edit"
        />
      </td>
      <td>
        <ClickIcon
          href="#add-investments-modal"
          setFunction={setCurrent}
          val={startup}
          iconName="attach_money"
        />
      </td>
      {/* Display grey button with no functionality if some financials have already been added */}
      {startup.financials.length === 0 ? (
        <td>
          <ClickIcon
            href="#add-kpi-names-modal"
            setFunction={setCurrent}
            val={startup}
            iconName="insert_chart"
          />
        </td>
      ) : (
        <td>
          <a
            id="unclickableIcon"
            href="!#"
            className="secondary-content"
            onClick={(e) => e.preventDefault()}
          >
            <i className="material-icons grey-text">insert_chart</i>
          </a>
        </td>
      )}
      <td>
        <a
          id="deleteIcon"
          href="#!"
          onClick={(e) => {
            e.preventDefault();
            deleteStartup(startup.id);
          }}
          className="secondary-content"
        >
          <i className="material-icons blue-text">delete</i>
        </a>
      </td>
    </tr>
  );
};

StartupItem.propTypes = {
  startup: PropTypes.object.isRequired,
  deleteStartup: PropTypes.func.isRequired,
  setCurrent: PropTypes.func.isRequired,
};

// Inject App Level state, so that
// component updates upon an update of info
// (e.g. added investment)
const mapStateToProps = (state) => ({
  startups: state.startup.startups,
});

export default connect(mapStateToProps, { setCurrent, deleteStartup })(
  StartupItem
);
