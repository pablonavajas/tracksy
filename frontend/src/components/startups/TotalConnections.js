import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import {getConnections} from  '../../actions/connectionsActions';


const TotalConnections = ({ current, connections, getConnections }) => {
    const [name, setName] = useState("");
    const [website, setWebsite] = useState("");

    useEffect(() => {
    getConnections();
  }, [connections, current]);

  if (connections === null) {
    return (
      <div className="progress">
        <div className="indeterminate"></div>
      </div>
    );
  }

  return (
  // Section: Stats
  
  <section className="section section-stats center">
    <nav>
    <div className="nav-wrapper blue-grey darken-4">
      <ul id="nav-mobile" className="left hide-on-med-and-down">
        <li><a href="#startup-overview">Overview</a></li>
        <li><a href="#hiring">Hiring</a></li>
        <li className="active"><a href="#total-connections">Connections</a></li>
      </ul>
    </div>
  </nav>
  
  <div className="col s12 m6 l4">
          <div className="card">
            <div className="card-content">
              <table className="striped">
                <thead>
                  <tr className = "center">
                    <th>Name</th>
                    <th>Description</th>
                    <th>URL</th>
                  </tr>
                </thead>
                <tbody>
                   {connections.map((c) => (
                     <tr key={c.id}>
                       <td> {c.name} </td>
                       <td> {c.description} </td>
                       <td> {c.url} </td>
                   </tr>
                   ))}
                </tbody>
              </table>
            </div>
        </div>
        </div>

      </section>
  );
};


TotalConnections.propTypes = {
  current: PropTypes.object,
  getConnections:PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  connections: state.connections,
  current: state.startup.current
});

export default connect(mapStateToProps, { getConnections })(TotalConnections);
