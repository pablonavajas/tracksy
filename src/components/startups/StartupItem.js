import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { deleteStartup, setCurrent } from '../../actions/startupsActions';

import M from 'materialize-css/dist/js/materialize.min.js';

const startupItem = ({ startup, setCurrent, deleteStartup }) => {
  const onDelete = () => {
    deleteStartup(startup.id);
    M.toast({ html: 'Startup Deleted' });
  };

  return (
    <tr>
      <th className="center">
        <a href={'https://' + startup.website}>
          <img
            src={'http://logo.clearbit.com/' + startup.website}
            alt=""
            width="100"
            height="100"
          />
        </a>
        <div>{startup.name}</div>
      </th>
      <td className="center">{startup.ownership}</td>
      <td className="center">{startup.board}</td>
      <td>{startup.investment_1}</td>
      <td className="center">{startup.type_1}</td>
      <td>{startup.date_1}</td>
      <td className="center">{startup.investment_2}</td>
      <td className="center">{startup.type_2}</td>
      <td>{startup.date_2}</td>
      <td>
        <a
          href="#edit-startup-modal"
          onClick={() => setCurrent(startup)}
          className="secondary-content modal-trigger"
        >
          <i className="material-icons grey-text">edit</i>
        </a>
      </td>
      <td width="50">
        <a href="#!" onClick={onDelete} className="secondary-content">
          <i className="material-icons grey-text">delete</i>
        </a>
      </td>
    </tr>
  );
};

startupItem.propTypes = {
  startup: PropTypes.object.isRequired,
  deleteStartup: PropTypes.func.isRequired,
  setCurrent: PropTypes.func.isRequired
};

export default connect(null, { setCurrent, deleteStartup })(startupItem);
