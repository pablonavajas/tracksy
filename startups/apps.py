from django.apps import AppConfig


class TracksyappConfig(AppConfig):
    name = 'startups'
