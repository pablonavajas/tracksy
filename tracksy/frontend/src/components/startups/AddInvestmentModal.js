import React, { useState, useEffect } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import CurrencyFormat from "react-currency-format";

import {
  deleteInvestment,
  addInvestments,
} from "../../actions/investmentsActions";
import { setCurrent } from "../../actions/startupsActions";

const addInvestmentModal = ({
  current,
  addInvestments,
  deleteInvestment,
  setCurrent,
}) => {
  const [currentName, setCurrentName] = useState("");
  const [list, setList] = useState([]);

  useEffect(() => {
    if (current) {
      setCurrentName(current.name);
      setList(current.investments);
    }
  }, [current]);

  const initialInputState = {
    value: "",
    currency: "£",
    investmentType: "",
    date: "",
  };

  const addInput = () => {
    setList([...list, initialInputState]);
  };

  const onSubmit = () => {
    addInvestments(current.id, list);
    setCurrent(null);
  };

  // 0-based index
  const deleteInput = (index) => {
    // delete the inputs from database if they were already there
    if (list[index].id) {
      deleteInvestment(current.id, list[index].id);
    }

    //delete the inputs from the modal
    list.splice(index, 1);
    setList([...list]);
  };

  return (
    <div id="add-investments-modal" className="modal" style={modalStyle}>
      <div>
        <div className="modal-content">
          <h4>Investments for {currentName}</h4>
          <div className="center">
            {list.map((inv, i) => (
              <div className="row" key={i}>
                <div className="input-field col s1">
                  <select
                    id={"invCurrency" + i}
                    name="currency"
                    value={inv.currency ? inv.currency : ""}
                    className="browser-default"
                    onChange={(e) => {
                      list[i].currency = e.target.value;
                      setList([...list]);
                    }}
                  >
                    <option value="£" defaultValue="£">
                      £
                    </option>
                    <option value="$">$</option>
                    <option value="€">€</option>
                  </select>
                  <label htmlFor={"invCurrency" + i} className="active">
                    Currency
                  </label>
                </div>
                <div className="input-field col s4" key={i}>
                  <CurrencyFormat
                    id={"invValue" + i}
                    value={inv.value}
                    thousandSeparator={true}
                    className="validate"
                    prefix={inv.currency ? inv.currency : "£"}
                    allowNegative={false}
                    onValueChange={(values) => {
                      const { formattedValue, value } = values;
                      // formattedValue = $2,223
                      // value ie, 2223
                      list[i].value = value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"inveValue" + i} className="active">
                    Investment
                  </label>
                </div>
                <div className="input-field col s4">
                  <input
                    placeholder=""
                    id={"invType" + i}
                    name="type_1"
                    type="text"
                    className="validate"
                    value={inv.investmentType ? inv.investmentType : ""}
                    onChange={(e) => {
                      list[i].investmentType = e.target.value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"invType" + i} className="active">
                    Type of Investment
                  </label>
                </div>
                <div className="input-field col s2">
                  <input
                    id={"invDate" + i}
                    name="date_1"
                    value={inv.date ? inv.date : ""}
                    type="date"
                    className="validate"
                    onChange={(e) => {
                      list[i].date = e.target.value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"invDate" + i} className="active">
                    Close Date
                  </label>
                  <a href="#" onClick={() => deleteInput(i)}>
                    <i className="material-icons grey-text prefix">
                      delete_forever
                    </i>
                  </a>
                </div>
              </div>
            ))}

            <a
              className="btn waves-effect waves-light light-blue"
              onClick={addInput}
            >
              Add Investment
            </a>
          </div>
        </div>
        <div className="modal-footer">
          <div className="center">
            <a
              onClick={onSubmit}
              className="modal-close waves-effect light-blue waves-light btn"
            >
              Submit
            </a>
          </div>
        </div>
      </div>
    </div>
  );
};

const modalStyle = {
  width: "60%",
  height: "75%",
};

addInvestmentModal.propTypes = {
  addInvestments: PropTypes.func.isRequired,
  deleteInvestment: PropTypes.func.isRequired,
  setCurrent: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  current: state.startup.current,
});

export default connect(mapStateToProps, {
  addInvestments,
  deleteInvestment,
  setCurrent,
})(addInvestmentModal);
