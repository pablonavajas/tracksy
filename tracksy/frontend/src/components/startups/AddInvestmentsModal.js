import React, { useState, useEffect, Fragment } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { addInvestment } from "../../actions/investmentsActions";
import CurrencyFormat from "react-currency-format";
import { setCurrent } from "../../actions/startupsActions";

const addInvestmentModal = ({ addInvestment, current }) => {
  const [currentName, setCurrentName] = useState("Hello");
  const [list, setList] = useState([
    {
      startupId: null,
      value: 0,
      currency: "£",
      investmentType: "",
      date: ""
    }
  ]);

  useEffect(() => {
    if (current) {
      setCurrentName(current.name);
      setList(current.investments);
    }
  }, [current]);

  const addInput = () => {
    list.push({
      startupId: current.id,
      value: "",
      currency: "£",
      investmentType: "",
      date: ""
    });
    setList([...list]);
  };

  const onSubmit = () => {
    list.map(inv => {
      addInvestment(inv);
    });

    // Clear fields
    // setList([]);
    setList([
      {
        startupId: null,
        value: 0,
        currency: "£",
        investmentType: "",
        date: ""
      }
    ]);
  };

  return (
    <div id="add-investments-modal" className="modal" style={modalStyle}>
      <div className="modal-content">
        <h4>Enter Investments for {currentName}</h4>
        <div className="center">
          <div className="row">
            {list.map((inv, i) => (
              <div key={i}>
                <div className="input-field col s1">
                  <select
                    id={"invCurrency" + i}
                    name="currency"
                    value={inv.currency}
                    className="browser-default"
                    onChange={e => {
                      list[i].currency = e.target.value;
                      setList([...list]);
                    }}
                  >
                    <option value="£" defaultValue="£">
                      £
                    </option>
                    <option value="$">$</option>
                    <option value="€">€</option>
                  </select>
                  <label htmlFor={"invCurrency" + i} className="active">
                    Currency
                  </label>
                </div>
                <div className="input-field col s4" key={i}>
                  <CurrencyFormat
                    id={"invValue" + i}
                    value={inv.value}
                    thousandSeparator={true}
                    prefix={inv.currency}
                    allowNegative={false}
                    onValueChange={values => {
                      const { formattedValue, value } = values;
                      // formattedValue = $2,223
                      // value ie, 2223
                      list[i].startupId = current.id;
                      list[i].value = value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"inveValue" + i} className="active">
                    Investment
                  </label>
                </div>
                <div className="input-field col s4">
                  <input
                    placeholder=""
                    id={"invType" + i}
                    name="type_1"
                    type="text"
                    value={inv.investmentType}
                    onChange={e => {
                      list[i].investmentType = e.target.value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"invType" + i} className="active">
                    Type of Investment
                  </label>
                </div>
                <div className="input-field col s3">
                  <input
                    id={"invDate" + i}
                    name="date_1"
                    value={inv.date}
                    type="date"
                    onChange={e => {
                      list[i].date = e.target.value;
                      setList([...list]);
                    }}
                  />
                  <label htmlFor={"invDate" + i} className="active">
                    Close Date
                  </label>
                </div>
              </div>
            ))}
          </div>
          <a
            className="btn waves-effect waves-light light-blue"
            onClick={addInput}
          >
            Add Investment
          </a>
        </div>
      </div>
      <div className="modal-footer">
        <a
          onClick={onSubmit}
          className="modal-close waves-effect blue waves-light btn-flat"
        >
          Submit
        </a>
      </div>
    </div>
  );
};

const modalStyle = {
  width: "60%",
  height: "75%"
};

addInvestmentModal.propTypes = {
  addInvestment: PropTypes.func.isRequired
};

const mapStateToProps = state => {
  // console.log(state.startup.current);
  return {
    current: state.startup.current
  };
};

export default connect(mapStateToProps, { addInvestment })(addInvestmentModal);
