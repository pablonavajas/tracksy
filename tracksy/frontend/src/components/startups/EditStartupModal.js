import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { updateStartup } from "../../actions/startupsActions";
//import moment from "moment/src/moment";
//var moment = require("moment");

const EditStartupModal = ({ current, updateStartup }) => {
  const [name, setName] = useState("");
  const [website, setWebsite] = useState("");
  const [ownership, setOwnership] = useState("");
  const [board, setBoard] = useState("");
  const [investment_1, setInvestment_1] = useState("");
  const [type_1, setType_1] = useState("");
  const [date_closed_1, setDate_1] = useState("");
  const [investment_2, setInvestment_2] = useState("");
  const [type_2, setType_2] = useState("");
  const [date_closed_2, setDate_2] = useState("");

  useEffect(() => {
    if (current) {
      setName(current.name);
      setWebsite(current.website);
      setOwnership(current.ownership);
      setBoard(current.board);
      setInvestment_1(current.investment_1);
      setDate_1(current.date_closed_1);
      setType_1(current.type_1);
      setInvestment_2(current.investment_2);
      setDate_2(current.date_closed_2);
      setType_2(current.type_2);
    }
  }, [current]);

  const onSubmit = () => {
    const updStartup = {
      id: current.id,
      name,
      website,
      ownership,
      board,
      investment_1,
      investment_2,
      type_1,
      type_2,
      date_closed_1,
      date_closed_2
    };

    updateStartup(updStartup);

    // Clear fields
    setName("");
    setWebsite("");
    setOwnership("");
    setBoard("");
    setInvestment_1("");
    setDate_1("");
    setType_1("");
    setInvestment_2("");
    setDate_2("");
    setType_2("");
  };

  return (
    <div id="edit-startup-modal" className="modal" style={modalStyle}>
      <div className="modal-content">
        <h4>Edit Startup</h4>
        {/** Name */}
        <div className="row">
          <div className="input-field">
            <input
              type="text"
              name="name"
              value={name}
              onChange={e => setName(e.target.value)}
            />
            <span className="helper-text">Startup Name</span>
          </div>
        </div>

        {/** Website */}
        <div className="row">
          <div className="input-field">
            <input
              type="text"
              name="website"
              value={website}
              onChange={e => setWebsite(e.target.value)}
            />
            <span className="helper-text">Website</span>
          </div>
        </div>

        {/** Ownership */}
        <div className="row">
          <div className="input-field">
            <input
              name="ownership"
              type="text"
              value={ownership ? ownership : ""}
              onChange={e => setOwnership(e.target.value)}
            />
            <span className="helper-text">Ownership (%)</span>
          </div>
        </div>

        {/** Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              name="investment_1"
              type="text"
              value={investment_1 ? investment_1 : ""}
              onChange={e => setInvestment_1(e.target.value)}
            />
            <span className="helper-text">Investment 1</span>
          </div>
        </div>
        {/** Type of Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              name="type_1"
              type="text"
              value={type_1 ? type_1 : ""}
              onChange={e => setType_1(e.target.value)}
            />
            <span className="helper-text">Type of investment 1</span>
          </div>
        </div>
        {/** Closing Date Investment 1 */}
        <div className="row">
          <div className="input-field">
            <input
              name="date_1"
              value={date_closed_1 ? date_closed_1 : ""}
              type="date"
              onChange={e => setDate_1(e.target.value)}
            />
            <span className="helper-text">Close Date</span>
          </div>
        </div>

        {/** Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              name="investment_2"
              type="text"
              value={investment_2 ? investment_2 : ""}
              onChange={e => setInvestment_2(e.target.value)}
            />
            <span className="helper-text">Investment 2</span>
          </div>
        </div>

        {/** Type of Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              name="type_2"
              type="text"
              value={type_2 ? type_2 : ""}
              onChange={e => setType_2(e.target.value)}
            />
            <span className="helper-text">Type of investment 2</span>
          </div>
        </div>

        {/** Closing Date Investment 2 */}
        <div className="row">
          <div className="input-field">
            <input
              name="date_2"
              value={date_closed_2}
              type="date"
              onChange={e => setDate_2(e.target.value)}
            />
            <span className="helper-text">Close Date</span>
          </div>
        </div>
      </div>
      <div className="row">
        <div className="input-field col s12">
          <select
            name="board"
            value={board ? board : ""}
            className="browser-default"
            onChange={e => setBoard(e.target.value)}
          >
            <option value="" disabled>
              Select Board
            </option>
            <option value="Member">Member</option>
            <option value=" - "> - </option>
            <option value="Observer"> Observer </option>
          </select>
        </div>
      </div>
      <div className="modal-footer">
        <a
          href="#!"
          onClick={onSubmit}
          className="modal-close waves-effect blue waves-light btn"
        >
          Submit
        </a>
      </div>
    </div>
  );
};

const modalStyle = {
  width: "60%",
  height: "75%"
};

EditStartupModal.propTypes = {
  current: PropTypes.object,
  updateStartup: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  current: state.startup.current
});

export default connect(mapStateToProps, { updateStartup })(EditStartupModal);
