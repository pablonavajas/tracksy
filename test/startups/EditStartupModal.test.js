import React from "react";
import { mount, shallow } from "enzyme";
import "../setupTests";
import toJson from "enzyme-to-json";
import { EditStartupModal } from "../../tracksy/frontend/src/components/startups/EditStartupModal";
import ConnectedEditStartupModal from "../../tracksy/frontend/src/components/startups/EditStartupModal";

import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
const mockStore = configureMockStore([thunk]);

const testStartup = {
  id: 1,
  name: "test",
  website: "test",
  ownership: "test",
  board: "test",
  startupEmail: "test",
};

// Testing MapStateToProps
describe("EditStartupModal ", () => {
  const mockFn = jest.fn();
  const current = {
    id: 1,
    name: "test",
  };

  const initialState = {
    startup: { current },
  };

  const store = mockStore(initialState);
  const wrapper = shallow(
    <ConnectedEditStartupModal updateStartup={mockFn} store={store} />
  );

  it("should show the current value imported from store", () => {
    expect(wrapper.props().children.props.current).toEqual(current);
  });
});

//Snapshot Component Test
describe("EditStartupModal ", () => {
  const mockFn = jest.fn();

  it("renders correctly", () => {
    const wrapper = shallow(<EditStartupModal updateStartup={mockFn} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

//Logic Component Test
describe("When all fields are filled in, in the EditStartupModal", () => {
  let wrapper;
  const mockFn = jest.fn();
  const mockEvent = {
    target: {
      name: "description",
      value: "test",
    },
  };

  beforeEach(() => {
    wrapper = mount(
      <EditStartupModal updateStartup={mockFn} current={testStartup} />
    );
    // fill in the input fields
    wrapper.find("#editStartupName").at(1).simulate("change", mockEvent);
    wrapper.find("#editWebsiteUrl").at(1).simulate("change", mockEvent);
    wrapper.find("#editOwnership").at(1).simulate("change", mockEvent);
    wrapper.find("#editStartupEmail").at(1).simulate("change", mockEvent);
    wrapper.find("#editBoard").at(1).simulate("change", mockEvent);
  });

  it("the props for all values have been correctly set to expected values", () => {
    expect(wrapper.find("#editStartupName").at(1).prop("value")).toEqual(
      "test"
    );
    expect(wrapper.find("#editWebsiteUrl").at(1).prop("value")).toEqual("test");
    expect(wrapper.find("#editOwnership").at(1).prop("value")).toEqual("test");
    expect(wrapper.find("#editStartupEmail").at(1).prop("value")).toEqual(
      "test"
    );
    expect(wrapper.find("#editBoard").at(1).prop("value")).toEqual("test");
  });

  it("and  submit button clicked, editStartup function is called with correct values", () => {
    wrapper.find("a").simulate("click");
    expect(mockFn).toHaveBeenCalledWith(testStartup);
  });
});
