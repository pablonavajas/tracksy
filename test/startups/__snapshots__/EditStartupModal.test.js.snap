// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditStartupModal  renders correctly 1`] = `
<div
  className="modal"
  id="edit-startup-modal"
  style={
    Object {
      "height": "75%",
      "width": "60%",
    }
  }
>
  <div
    className="modal-content"
  >
    <h4>
      Edit Startup
    </h4>
    <div
      className="row"
    >
      <InputField
        id="editStartupName"
        name="Startup Name"
        placeholder=""
        setFunction={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="row"
    >
      <InputField
        id="editWebsiteUrl"
        name="Website URL"
        placeholder=""
        setFunction={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="row"
    >
      <InputField
        id="editOwnership"
        name="Ownership (%)"
        placeholder=""
        setFunction={[Function]}
        type="number"
        value=""
      />
    </div>
    <div
      className="row"
    >
      <InputField
        id="editStartupEmail"
        name="Startup Email"
        placeholder=""
        setFunction={[Function]}
        type="email"
        value=""
      />
    </div>
    <div
      className="row"
    >
      <SelectField
        id="editBoard"
        name="Board"
        options={
          Array [
            " - ",
            "Member",
            "Observer",
          ]
        }
        setFunction={[Function]}
        value=""
      />
    </div>
  </div>
  <div
    className="modal-footer"
  >
    <div
      className="center"
    >
      <a
        className="modal-close waves-effect light-blue waves-light btn"
        onClick={[Function]}
      >
        Submit
      </a>
    </div>
  </div>
</div>
`;
